### PR TITLE
Stubs and getters/setters for atom fires.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Procs/varname changes to support #4644

### CODE REVIEW

File: code/game/machinery/igniter.dm [hunk 1]
```diff
@@ -51,7 +51,7 @@
 			location.hotspot_expose(1000,500,1)
 	return 1
 
-/obj/machinery/igniter/proc/ignite()
+/obj/machinery/igniter/proc/create_sparks()
```
Comment (out-of-phaze) [inline code/game/machinery/igniter.dm]: this doesn't actually produce sparks at all

File: code/game/objects/items/flame/_flame.dm [hunk 2]
```diff
@@ -216,17 +216,14 @@
...
-		loc.update_icon()
-
-	// TODO: generalized ignition proc
-	if(isliving(loc))
-		var/mob/living/M = loc
-		M.IgniteMob()
+	if(loc)
+		if(istype(loc, /obj/structure/wall_sconce))
+			loc.update_icon()
+		loc.ignite_fire()
```
Comment (out-of-phaze) [inline code/game/objects/items/flame/_flame.dm:220]: could put an icon update in sconces' ignite_fire?

File: code/game/objects/items/flame/_flame.dm [hunk 3]
```diff
@@ -115,7 +115,7 @@
 
 	return ..()
 
-/obj/item/flame/proc/extinguish(var/mob/user, var/no_message)
+/obj/item/flame/extinguish_fire(mob/user, no_message = FALSE)
```
Comment (out-of-phaze) [inline code/game/objects/items/flame/_flame.dm]: not sure how to feel about this. i feel like a lighter should be able to catch fire separately from it being on/off or lit/unlit? idk

**Review by out-of-phaze — DISMISSED** [review]
i feel like "does this have the ability to be lit, a la lighters" and "is the material this is made of *on fire?*" should be separate, conceptually, but I'm not hard tied to that

File: code/game/machinery/igniter.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/game/machinery/igniter.dm]: Oh true, I assumed. Will rename it to `toggle_igniter` or something.

File: code/game/objects/items/flame/_flame.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/game/objects/items/flame/_flame.dm:220]: This actually might cause a recursion issue if the sconce then tries to light the torch again.

File: code/game/objects/items/flame/_flame.dm (hunk 3 above)
Comment (MistakeNot4892) [reply code/game/objects/items/flame/_flame.dm]: I wanted to condense all the procs but I was iffy on this too, I'll give it its own proc.

File: code/game/objects/items/flame/_flame.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/game/objects/items/flame/_flame.dm:220]: Looks like sconces don't try to light their contents via that proc so should be fine.

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**